front/src/components/boxs/device-in-room/EditDevices.jsx

```diff
@@ -83,7 +83,15 @@ class EditDevices extends Component {
   }
 
   componentWillReceiveProps(nextProps) {
-    this.refreshSelectedOptions(nextProps);
+    if (nextProps.box && nextProps.box.device_features) {
+      if (
+        !this.props.box ||
+        !this.props.box.device_features ||
+        nextProps.box.device_features.length !== this.props.box.device_features.length
```

Comment:
Adding a check on the length will not fix the issue, if I have 2 widgets with exactly the same number of deviceFeatures but different one, this will not be triggered and it should be